runtime-modules/project-token/src/lib.rs

```diff
@@ -0,0 +1,388 @@
+
+    type MerkleProof = MerkleProofOf<T>;
+
+    /// Change to permissionless
+    /// Preconditions:
+    /// - Token `token_id` must exist
+    /// Postconditions
+    /// - transfer policy of `token_id` changed to permissionless
+    fn change_to_permissionless(token_id: T::TokenId) -> DispatchResult {
+        TokenInfoById::<T>::try_mutate(token_id, |token_info| {
+            token_info.transfer_policy = TransferPolicyOf::<T>::Permissionless;
+            Ok(())
+        })
+    }
+
+    /// Reduce patronage rate by amount
+    /// Preconditions:
+    /// - `token_id` must exists
+    /// - `decrement` must be less or equal than current patronage rate for `token_id`
+    ///
+    /// Postconditions:
+    /// - patronage rate for `token_id` reduced by `decrement`
+    fn reduce_patronage_rate_by(token_id: T::TokenId, decrement: T::Balance) -> DispatchResult {
+        let token_info = Self::ensure_token_exists(token_id)?;
+
+        // ensure new rate is >= 0
+        ensure!(
+            token_info.patronage_info.rate >= decrement,
+            Error::<T>::ReductionExceedingPatronageRate,
+        );
+
+        // == MUTATION SAFE ==
+
+        let now = Self::current_block();
+        let new_rate = token_info.patronage_info.rate.saturating_sub(decrement);
+        TokenInfoById::<T>::mutate(token_id, |token_info| {
+            token_info
+                .patronage_info
+                .set_new_rate_at_block::<T::BlockNumberToBalance>(new_rate, now);
+        });
+
+        Self::deposit_event(RawEvent::PatronageRateDecreasedTo(token_id, new_rate));
+
+        Ok(())
+    }
+
+    /// Allow creator to receive credit into his accounts
+    /// Preconditions:
+    /// - `token_id` must exists
+    /// - `to_account` must be valid for `token_id`
+    ///
+    /// Postconditions:
+    /// - outstanding patronage credit for `token_id` transferred to `to_account`
+    /// - outstanding patronage credit subsequently set to 0
+    /// no-op if outstanding credit is zero
+    fn claim_patronage_credit(token_id: T::TokenId, to_account: T::AccountId) -> DispatchResult {
+        let token_info = Self::ensure_token_exists(token_id)?;
+        Self::ensure_account_data_exists(token_id, &to_account).map(|_| ())?;
+
+        let now = Self::current_block();
+        let outstanding_credit = token_info
+            .patronage_info
+            .outstanding_credit::<T::BlockNumberToBalance>(now);
+
+        if outstanding_credit.is_zero() {
+            return Ok(());
+        }
+
+        // == MUTATION SAFE ==
+
+        AccountInfoByTokenAndAccount::<T>::mutate(token_id, &to_account, |account_info| {
+            account_info.increase_liquidity_by(outstanding_credit)
+        });
+
+        TokenInfoById::<T>::mutate(token_id, |token_info| {
+            token_info
+                .patronage_info
+                .reset_tally_at_block::<T::BlockNumberToBalance>(now);
+            token_info.increase_issuance_by(outstanding_credit);
```

Comment:
Perhaps we should bundle tally update inside `increase_issuance_by` as one always want to do this together? Same in `decrease_issuance_by`.

Reply:
This is fine as long as minting is tied to patronage, but wouldn't it cause problems with AMM bonding?

Reply:
addressed here: cefb6b18bf4e161c0c2512591d0fe0cd805b21b4

Reply:
I did not understand the problem here.

Reply:
I just want `increase_token_issuance_by(amount)` to just increase the supply and nothing else